tf_fastmri_data/datasets/non_cartesian.py

```diff
@@ -93,14 +93,14 @@ def preprocessing(self, image, kspace):
             )
         traj = tf.repeat(traj, tf.shape(image)[0], axis=0)
         orig_image_channels = ortho_ifft2d(kspace)
+        if self.crop_image_data:
+            image = adjust_image_size(image, self.image_size)
         nc_kspace = nufft(self.nufft_obj, orig_image_channels, traj, self.image_size, multicoil=self.multicoil)
         nc_kspace, image = scale_tensors(nc_kspace, image, scale_factor=self.scale_factor)
         image = image[..., None]
         nc_kspaces_channeled = nc_kspace[..., None]
         orig_shape = tf.ones([tf.shape(kspace)[0]], dtype=tf.int32) * self.image_size[-1]
-        if self.crop_image_data:
-            image = adjust_image_size(image, self.image_size)
-        else:
+        if not self.crop_image_data:
             output_shape = tf.shape(image)[1:][None, :]
```

Comment:
Won't this be a problem now because `tf.shape(image)[1:]` is going to be a tensor of length 3?

I mean not a problem per se, because we slice anyway in the network, but it carries extra unnecessary info.
Maybe you could do this `if` a bit earlier (like as the `else` of the modification above)

Reply:
I dont think it will.. I had to do this mainly cause the line `image = image[..., None]` would make the size bigger than what is needed by input for `adjust_image_size` function. Rest of the code is pretty much the same.

Reply:
honestly I am fine with it nv mind